robotoff/app/api.py

```diff
@@ -233,38 +230,38 @@ def on_post(self, req: falcon.Request, resp: falcon.Response):
         self.spellcheck(req, resp)
 
     def spellcheck(self, req: falcon.Request, resp: falcon.Response):
-        text = req.get_param("text")
-        barcode = req.get_param("barcode")
+        text = self.__get_text(req, resp)
+        if text is None:
+            return
+
         index_name = req.get_param(
             "index", default=settings.ELASTICSEARCH_PRODUCT_INDEX
         )
         confidence = req.get_param_as_float("confidence", default=1.0)
+        spellchecker = Spellchecker(
+            client=es_client, index_name=index_name, confidence=confidence
+        )
 
-        if text is None and barcode is None:
-            raise falcon.HTTPBadRequest("text or barcode is required.")
-
-        if not text:
-            product = get_product(barcode) or {}
-            text = product.get("ingredients_text_fr")
+        resp.media = {}
+        resp.media["status"] = "success"
+        resp.media["text"] = text
+        resp.media["corrected"] = spellchecker.correct(text)
+        resp.media["corrections"] = spellchecker.get_corrections()
 
-            if not text:
-                resp.media = {
-                    "status": "not_found",
-                }
-                return
+    def __get_text(self, req: falcon.Request, resp: falcon.Response) -> Optional[str]:
```

Comment:
I think it would be easier to read if __get_text body was directly at the beginning of spellcheck function.

Reply:
Useless method indeed. I added it to avoid duplicated code when I splitted the Spellcheck API into V1 and V2 routes. But I then removed the old version.
https://github.com/openfoodfacts/robotoff/pull/181/commits/1eb01f25d7a641bd2bf628bd2a45b966808a9e8b